cachecleaner.py

```diff
@@ -60,31 +63,40 @@ def clean_cache(workdir, capacity, quiet=False, time_type='st_atime'):
         return []
 
     with section('Sorting files', quiet) as write:
-        files.sort(reverse=True)
+        files.sort()
 
-        oldest = datetime.utcnow() - datetime.utcfromtimestamp(files[-1][0])
+        oldest = datetime.utcnow() - datetime.utcfromtimestamp(files[0][0])
         write('oldest file: {}'.format(oldest))
 
-        total_size = 0
-        for i, (_, size, _) in enumerate(files):
-            total_size += size
-            if total_size > capacity:
-                break
-        files = files[i:]
-
-        lifetime = datetime.utcnow() - datetime.utcfromtimestamp(files[0][0])
-        write('cache lifetime: {}'.format(lifetime))
-
     with section('Deleting files', quiet) as write:
-        for _, _, f in tqdm(files, disable=quiet, bar_format=BAR_FORMAT):
-            try:
-                os.remove(workdir + f)
-            except OSError:
-                continue
+        skipped = deleted = deleted_size = 0
+        with tqdm(total=total_size - capacity, disable=quiet,
+                  bar_format=BAR_FORMAT, unit='b',
+                  unit_scale=True, unit_divisor=1024) as bar:
+            for atime, size, fn in files:
+                fn = workdir + fn
+                if total_size - deleted_size <= capacity:
+                    break
+
+                try:
+                    stats = os.stat(fn)
+                    if getattr(stats, time_type) > atime:
+                        skipped += 1
+                        continue
+                    os.remove(fn)
+                except OSError:
+                    skipped += 1
+                    continue
+                update_bar(bar, size)
+                deleted_size += size
+                deleted += 1
+
+        oldest = datetime.utcnow() - datetime.utcfromtimestamp(atime)
+        write('oldest file: {}'.format(oldest))
         write('deleted: {} files, {:0.1f} mb'.format(
-            len(files),
-            sum(stats[1] for stats in files) / MEGABYTE,
+            deleted, deleted_size / MEGABYTE,
         ))
+        write('skipped: {} files'.format(skipped))
```

Comment:
it will be useful to know how deep we scan files, 10/1000, 999/1000

```
write('scaned: {} files from {}'.format(deleted + skipped, len(files))
```

Reply:
The total amount of files in the cache is in the first section and on the progress bar in the second section.

`skipped` already here and is usual 0, sometimes up to ten. `deleted` already here. In conclusion we'll have something like this every time:

```
Deleting files...
    100%|█████████████████████| 10.0M/10.0M [00:00<00:00, 15.0Mb/s]
    deleted: 4145 files, 10.0 mb
    skipped: 3 files
    scaned: 4148 files from 76607
    took 0.70 sec
```

I believe this is not informative.